torch/_torch_docs.py

```diff
@@ -2563,31 +2563,40 @@ def merge_dicts(*dicts):
 
 add_docstr(torch.dot,
            r"""
-dot(input, tensor) -> Tensor
+dot(input, other, *, out=None) -> Tensor
 
-Computes the dot product (inner product) of two tensors.
+Computes the dot product (inner product) of two 1D tensors.
 
-.. note:: This function does not :ref:`broadcast <broadcasting-semantics>`.
+.. note:: This function is similar to numpy.dot only for 1D tensors.
+
+Args:
+    input (Tensor): first tensor in the dot product, must be 1D.
+    other (Tensor): second tensor in the dot product, must be 1D.
+
+Keyword args:
+    {out}
 
 Example::
 
     >>> torch.dot(torch.tensor([2, 3]), torch.tensor([2, 1]))
     tensor(7)
+    >>> torch.dot(torch.tensor([2, 3]), torch.tensor([2]))
+    tensor([10])
 """)
 
 add_docstr(torch.vdot,
            r"""
 vdot(input, other, *, out=None) -> Tensor
 
-Computes the dot product (inner product) of two tensors. The vdot(a, b) function
+Computes the dot product (inner product) of two 1D tensors. The vdot(a, b) function
 handles complex numbers differently than dot(a, b). If the first argument is complex,
 the complex conjugate of the first argument is used for the calculation of the dot product.
 
-.. note:: This function does not :ref:`broadcast <broadcasting-semantics>`.
+.. note:: This function is similar to numpy.vdot only for 1D tensors.
```

Comment:
.. note:: Unlike NumPy's `vdot`, `torch.vdot` intentionally only supports computing the dot product of two 1D tensors with the same number of elements.

Reply:
Same wording suggestion for the previous documentation, too.